### ISSUE TITLE
[BUG] Creating a columns using np.nan throws error 

### ISSUE THREAD

**Issue by AK-ayush** [rapidsai/cudf#628]
**Describe the bug**
Creating a columns using np.nan throws error 

**Steps/Code to reproduce bug**
```
>>> from cudf.dataframe import DataFrame, Series
>>> s = Series([5, 1, 8, np.nan, 4, np.nan, 3, 7, np.nan, 3, 1, np.nan])
...
/conda/envs/cudf_add/lib/python3.5/site-packages/pyarrow/array.pxi in pyarrow.lib.array()

/conda/envs/cudf_add/lib/python3.5/site-packages/pyarrow/array.pxi in pyarrow.lib._sequence_to_array()

/conda/envs/cudf_add/lib/python3.5/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowInvalid: cannot convert float NaN to integer
```
Other examle:
```
>>> gdf = DataFrame([('a', list(range(5))),('b', list(range(5))),('c', list(range(5)))])
>>> gdf['d']=np.nan
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-47-3b9c4f9c0302> in <module>
----> 1 gdf['d']=5*[np.nan]

/conda/envs/cudf_add/lib/python3.5/site-packages/cudf-0+unknown-py3.5-linux-x86_64.egg/cudf/dataframe/dataframe.py in __setitem__(self, name, col)
    248             self._cols[name] = self._prepare_series_for_add(col)
    249         else:
--> 250             self.add_column(name, col)
    251
    252     def __delitem__(self, name):

/conda/envs/cudf_add/lib/python3.5/site-packages/cudf-0+unknown-py3.5-linux-x86_64.egg/cudf/dataframe/dataframe.py in add_column(self, name, data, forceindex)
    597             raise NameError('duplicated column name {!r}'.format(name))
    598
--> 599         series = self._prepare_series_for_add(data, forceindex=forceindex)
    600         series.name = name
    601         self._cols[name] = series

/conda/envs/cudf_add/lib/python3.5/site-packages/cudf-0+unknown-py3.5-linux-x86_64.egg/cudf/dataframe/dataframe.py in _prepare_series_for_add(self, col, forceindex)
    571         """
    572         self._sanitize_columns(col)
--> 573         col = self._sanitize_values(col)
    574
    575         empty_index = len(self._index) == 0

/conda/envs/cudf_add/lib/python3.5/site-packages/cudf-0+unknown-py3.5-linux-x86_64.egg/cudf/dataframe/dataframe.py in _sanitize_values(self, col)
    555             return Series(arr)
    556         elif len(self) > 0 and len(sind) != len(index):
--> 557             raise ValueError('Length of values does not match index length')
    558         return col
    559

ValueError: Length of values does not match index length
```
I am using latest cudf build from `branch-0.5`

**Comment by kkraus14** [comment]
This is a tough situation, because Arrow has explicitly stated they're going to error on mixed dtype situations like these, but we'd ideally want to support it. I'll look into falling back to a numpy array when it excepts on creating an Arrow array for situations like these.

**Comment by kkraus14** [comment]
Fixed by #857
